nixpkgs_merge_bot/github/GitHubClient.py

```diff
@@ -220,16 +223,19 @@ def create_issue_reaction(
                 )
 
     def merge_pull_request(
-        self, owner: str, repo: str, pr_number: int, sha: str
+        self, owner: str, repo: str, pr_number: int, sha: str, commenter
     ) -> HttpResponse | None:
         global STAGING
         if STAGING:
-            log.debug("Staging, not merging")
+            log.debug(f"pull request {pr_number}: Staging, not merging")
             return None
-        else:
-            return self.put(
-                f"/repos/{owner}/{repo}/pulls/{pr_number}/merge", data={"sha": sha}
-            )
+
+        commit_message = f"Co-authored-by: {commenter['login']} <{commenter['email'] or commenter['id']}>"
```

Comment:
I think it makes sense to always include the id, since email addresses and usernames are subject to change

Reply:
Github links those issues via the email address as it is intended for the Co-authored-by field
https://docs.github.com/en/pull-requests/committing-changes-to-your-project/creating-and-editing-commits/creating-a-commit-with-multiple-authors#creating-co-authored-commits-on-github
So I would argue to stick to that